app/assets/javascripts/compute_resource.js

```diff
@@ -48,13 +49,16 @@ function testConnection(item) {
         $("#compute_resource_password").prop('disabled', false);
       }
       if (!/alert-danger/i.test(result)) {
-        notify("<p>" + __("Test connection was successful") + "</p>", 'success')
+        notify("<p>" + __("Test connection was successful") + "</p>", 'success');
+      } else {
+        $("#btn_datacenter").removeClass("btn-default").addClass("btn-danger");
```

Comment:
I can't see how to trigger this - it might be another bug, but alert-danger doesn't ever seem to be used in test responses?